src/storage/src/render/upsert/rocksdb.rs

```diff
@@ -24,51 +27,186 @@ pub const BATCH_SIZE: usize = 1024;
 /// A `UpsertState` implementation backed by RocksDB.
 /// This is currently untested, and simply compiles.
 pub struct RocksDB {
-    rocksdb: RocksDBInstance<UpsertKey, UpsertValue>,
+    inner: Inner,
+
+    // Bincode options and buffer used
+    // in `merge_snapshot_chunk`.
+    bincode_opts: BincodeOpts,
```

Comment:
Can you explain why the XOR details have to leak into the implementation of key/val stores? I was expecting that all this logic would be done externally and these implementation would only implement a dumb `multi_get`/`multi_put` API

Reply:
This is fixed in the newest commit!